File: actionbar/src/mobi/cangol/mobile/actionbar/ActionTabItem.java
package mobi.cangol.mobile.actionbar;

import android.os.Parcel;
import android.os.Parcelable;
/**
 * @author Cangol
 */
public  class ActionTabItem implements Parcelable{
    /**
     * 唯一标识
     */
	private int id=-1;
    /**
     * 标题文本
     */
	private String title=null;
    /**
     * 是否选中
     */
	private int selected=0;
	public ActionTabItem(){
		
	}
	
	public ActionTabItem(int id,String title, int selected) {
		super();
		this.id = id;
		this.title = title;
		this.selected = selected;
	}
	public int getId() {
		return id;
	}
	public void setId(int id) {
		this.id = id;
	}
	public String getTitle() {
		return title;
	}
	public void setTitle(String title) {
		this.title = title;
	}
	

	public int getSelected() {
		return selected;
	}

	public void setSelected(int selected) {
		this.selected = selected;
	}

	@Override
	public int describeContents() {
		return 0;
	}
	@Override
	public void writeToParcel(Parcel dest, int flags) {
		dest.writeInt(id);
		dest.writeString(title);
		dest.writeInt(selected);
	}

	public static final Parcelable.Creator<ActionTabItem> CREATOR = new Creator<ActionTabItem>() {

		@Override
		public ActionTabItem createFromParcel(Parcel source) {
			ActionTabItem p = new ActionTabItem();
			p.setId(source.readInt());
			p.setTitle(source.readString());
			p.setSelected(source.readInt());
			return p;
		}

		@Override
		public ActionTabItem[] newArray(int size) {
			return new ActionTabItem[size];
		}
	};
}

File: actionbar/src/main/java/mobi/cangol/mobile/actionbar/ActionTabItem.kt
package mobi.cangol.mobile.actionbar

import android.os.Parcel
import android.os.Parcelable

/**
 * @author Cangol
 */
class ActionTabItem() : Parcelable {
    /**
     * 唯一标识
     */
    var id = -1
    /**
     * 文本
     */
    var text: String? = null
    /**
     * 是否选中
     */
    var selected = 0

    constructor(parcel: Parcel) : this() {
        id = parcel.readInt()
        text = parcel.readString()
        selected = parcel.readInt()
    }

    constructor(id: Int, text: String, selected: Int) : this() {
        this.id = id
        this.text = text
        this.selected = selected
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(id)
        dest.writeString(text)
        dest.writeInt(selected)
    }

    companion object CREATOR : Parcelable.Creator<ActionTabItem> {
        override fun createFromParcel(parcel: Parcel): ActionTabItem {
            return ActionTabItem(parcel)
        }

        override fun newArray(size: Int): Array<ActionTabItem?> {
            return arrayOfNulls(size)
        }
    }


}
